content/blog/2025/04/05/2025-04-05-design-post.adoc

```diff
@@ -0,0 +1,116 @@
+---
+layout: post
+title: "Redesigning Jenkins (Part One)"
+tags:
+- jenkins
+- user-experience
+- ux
+authors:
+- janfaracik
+opengraph:
+  image: /images/post-images/2025/04/2025-04-05-design-post/opengraph.png
+---
+
+image::/images/post-images/2025/04/2025-04-05-design-post/banner.png[role=center]
+
+Jenkins is the most used CI/CD platform in the world. It's one of the oldest, most mature systems in
+continuous integration and delivery, and it powers software pipelines across critical industries like finance,
+healthcare, and government. With over a million lines of code, more than 2,000 plugins, and millions of users,
+Jenkins is less a "tool" and more of a global infrastructure backbone.
+
+And because of all that - making changes is hard. Really hard.
+
+== Part One - The Problem
+
+The web has evolved significantly since Jenkins' inception - and as a result of that we've been carrying a lot of
+legacy technology along the way.
+
+We’ve carried forward legacy technologies like http://prototypejs.org[Prototype.js], https://yui.github.io/yui2/[YUI 2],
+https://lesscss.org[LESS], fragments of https://jquery.com[early jQuery], and a patchwork of UI components from
+different eras of frontend development. The result? A frontend stack that's tough to maintain, painful to extend, and a
+real barrier for contributors.
+
+That's not a knock on Jenkins - it's just the reality of any large, long-lived platform. Over time, the
+ground shifts beneath you. What was best practice in 2005 becomes technical debt in 2025.
+
+So, we knew something had to change.
+
+=== Maintaining compatibility
+
+In modernizing the Jenkins frontend, our foremost priority has been to preserve backwards compatibility,
+maintain platform stability, and ensure that no production pipelines are disrupted as a result of UI or
+framework changes. Given Jenkins' role in powering mission-critical CI/CD workflows across global
+enterprises, even small regressions can have far-reaching impact.
+
+To achieve this, our approach has been pragmatic, cautious, and highly incremental. Rather than introducing
+disruptive overhauls, we have focused on gradually modernizing the codebase over a period of several years,
+ensuring each change is thoroughly tested and validated in real-world use cases.
```

Comment:
```suggestion
To achieve this, our approach has been pragmatic, cautious, and highly incremental.
Rather than introducing disruptive overhauls, we have focused on gradually modernizing the codebase over a period
of several years, ensuring each change is thoroughly tested and validated in real-world use cases.
```